workers/task-driver/src/tasks/node_startup.rs

```diff
@@ -0,0 +1,187 @@
+//! Node startup task; defines the process by which a node bootstraps into the
+//! network and (possibly) into an existing cluster's raft
+
+// --------------
+// | Task State |
+// --------------
+
+use std::{error::Error, fmt::Display, time::Duration};
+
+use arbitrum_client::client::ArbitrumClient;
+use async_trait::async_trait;
+use common::types::tasks::NodeStartupTaskDescriptor;
+use job_types::{network_manager::NetworkManagerQueue, proof_manager::ProofManagerQueue};
+use serde::Serialize;
+use state::{error::StateError, State};
+use tracing::{info, instrument};
+
+use crate::{
+    driver::StateWrapper,
+    traits::{Task, TaskContext, TaskError, TaskState},
+};
+
+/// The name of the node startup task
+const NODE_STARTUP_TASK_NAME: &str = "node-startup";
+
+/// Defines the state of the node startup task
+#[derive(Clone, Debug, PartialEq, Eq, PartialOrd, Ord)]
+pub enum NodeStartupTaskState {
+    /// The task is awaiting scheduling
+    Pending,
+    /// The task is waiting for the gossip layer to warm up
+    GossipWarmup,
+    /// The task is completed
+    Completed,
+}
```

Comment:
understand that this a skeleton, but do we wanna stub in the other steps now? e.g. `RaftCatchup`, `SetupRelayerWallet`?

Reply:
not sure what the state are yet so i'll hold off til implementation